### PR TITLE
JMS: Support other datatypes #199

### PR DESCRIPTION
**Pull request by d2e8f9**
- Introduces JMSMessage trait for Map-, Byte-, Text- and ObjectMessages and moved it to separate file
- Created Sink and Source creation methods for the new datatypes
- Added Test
- Updated Documentation

### CODE REVIEW

File: docs/src/main/paradox/jms.md [hunk 1]
```diff
@@ -40,256 +40,236 @@ Gradle
...
+Use a case class with the subtype of @scaladoc[JmsMessage](akka.stream.alpakka.jms.JmsMessage$) to wrap the messages you want to send and optionally set their properties (see below for an example).
 
 ### Sending messages to a JMS provider
 
-First define a jms @javadoc[javax.jms.ConnectionFactory](javax.jms.ConnectionFactory) depending on the implementation you're using. Here we're using Active MQ.
+First define a jms `ConnectionFactory` depending on the implementation you're using. Here we're using Active MQ.
 
 Scala
-: @@snip ($alpakka$/jms/src/test/scala/akka/stream/alpakka/jms/scaladsl/JmsConnectorsSpec.scala) { #connection-factory }
+: @@snip (../../../../jms/src/test/scala/akka/stream/alpakka/jms/scaladsl/JmsConnectorsSpec.scala) { #connection-factory }
```
Comment (ennru) [inline docs/src/main/paradox/jms.md]: replace all with absolute path `$alpakka$`, please

File: docs/src/main/paradox/jms.md [hunk 2]
```diff
@@ -40,256 +40,236 @@ Gradle
...
 
 Java
-: @@snip ($alpakka$/jms/src/test/java/akka/stream/alpakka/jms/javadsl/JmsConnectorsTest.java) { #connection-factory }
+: @@snip (../../../../jms/src/test/java/akka/stream/alpakka/jms/javadsl/JmsConnectorsTest.java) { #connection-factory }
 
 
-Create a sink, that accepts and forwards @javadoc[String](java.lang.String)s to the JMS provider.
+Create a sink, that accepts and forwards @extref[String](java-api:java.lang.String)s to the JMS provider.
```
Comment (ennru) [inline docs/src/main/paradox/jms.md]: Please use the newly configured `@javadoc[String](java.lang.String)` style links.

File: docs/src/main/paradox/jms.md [hunk 3]
```diff
@@ -40,256 +40,236 @@ Gradle
...
 ## Usage
 
-The JMS message model supports several types of message body (see @javadoc[javax.jms.Message](javax.jms.Message)) and Alpakka currently supports messages with a body containing a @javadoc[String](java.lang.String) object.
+The JMS message model supports several types of message body (see @extref(javaee-api:javax.jms.Message)) and Alpakka currently supports messages with a body containing a @extref[String](java-api:java.lang.String), 
+a @extref[Serializable](java-api:java.io.Serializable) object, a @extref[Map](java-api:java.util.Map) or a byte array.
 
-Use the case class @scaladoc[JmsTextMessage](akka.stream.alpakka.jms.JmsTextMessage$) to wrap the messages you want to send and optionally set their properties (see below for an example).
+Use a case class with the subtype of @scaladoc[JmsMessage](akka.stream.alpakka.jms.JmsMessage$) to wrap the messages you want to send and optionally set their properties (see below for an example).
 
 ### Sending messages to a JMS provider
```
Comment (ennru) [inline docs/src/main/paradox/jms.md]: Change to `##` levels for the sections show in the left hand overview.

File: jms/src/main/scala/akka/stream/alpakka/jms/Jms.scala [hunk 4]
```diff
@@ -4,113 +4,22 @@
...
 
-import scala.collection.JavaConverters._
-
 sealed trait JmsSettings {
   def connectionFactory: ConnectionFactory
+
   def destination: Option[Destination]
+
   def credentials: Option[Credentials]
-  def acknowledgeMode: AcknowledgeMode
```
Comment (ennru) [inline jms/src/main/scala/akka/stream/alpakka/jms/Jms.scala]: What happend to `AcknowledgeMode`?

File: docs/src/main/paradox/jms.md [hunk 5]
```diff
@@ -40,256 +40,236 @@ Gradle
...
+: @@snip (../../../../jms/src/test/scala/akka/stream/alpakka/jms/scaladsl/JmsConnectorsSpec.scala) { #assert-only-odd-messages-received }
 
 Java
-:   ```
-    sbt
-    > jms/testOnly *.JmsConnectorsTest
-    ```
+: @@snip (../../../../jms/src/test/java/akka/stream/alpakka/jms/javadsl/JmsConnectorsTest.java) { #assert-only-odd-messages-received }
 
-### Using IBM MQ
```
Comment (ennru) [inline docs/src/main/paradox/jms.md]: What happend to the IBM MQ section?

File: docs/src/main/paradox/jms.md (hunk 3 above)
Comment (d2e8f9) [reply docs/src/main/paradox/jms.md]: Documentation is restructured. Please have a look

File: docs/src/main/paradox/jms.md (hunk 1 above)
Comment (d2e8f9) [reply docs/src/main/paradox/jms.md]: Fixed

File: docs/src/main/paradox/jms.md (hunk 2 above)
Comment (d2e8f9) [reply docs/src/main/paradox/jms.md]: Fixed

File: docs/src/main/paradox/jms.md (hunk 5 above)
Comment (d2e8f9) [reply docs/src/main/paradox/jms.md]: Fixed. Maybe it was a little bit to late yesterday...

File: jms/src/main/scala/akka/stream/alpakka/jms/Jms.scala (hunk 4 above)
Comment (d2e8f9) [reply jms/src/main/scala/akka/stream/alpakka/jms/Jms.scala]: Fixed.

**Comment by ennru** [comment]
Thank you for your contribution!

**Pull request merged**